watson/frames.py

```diff
@@ -143,13 +143,28 @@ def new_frame(self, project, start, stop, tags=None, id=None,
     def dump(self):
         return tuple(frame.dump() for frame in self._rows)
 
-    def filter(self, projects=None, tags=None, span=None):
-        return (
-            frame for frame in self._rows
-            if (projects is None or frame.project in projects) and
-               (tags is None or any(tag in frame.tags for tag in tags)) and
-               (span is None or frame in span)
-        )
+    def filter(self, projects=None, tags=None, span=None,
+               include_partial_frames=False):
+        for frame in self._rows:
+            if not (projects is None or frame.project in projects):
```

Comment:
```suggestion
            if projects is not None and frame.project not in projects:
```

Would be easier to read IMO :) .